cpp/src/command_classes/SwitchBinary.cpp

```diff
@@ -107,7 +107,7 @@ namespace OpenZWave
 					// data[1] => Switch state
 					if (Internal::VC::ValueBool* value = static_cast<Internal::VC::ValueBool*>(GetValue(_instance, ValueID_Index_SwitchBinary::Level)))
 					{
-						if (GetVersion() >= 2)
+						if (GetVersion() >= 2 && _length == 4)
```

Comment:
I think the check should be:
```c++
if (GetVersion() >= 2 && _length >= 4)
```
If versions after 2 are ever introduced, the length will also increase. This would hold true for the code in `Basic.cpp` as well. See `Alarm.cpp` in contrast.

Reply:
That makes sense.  I was hoping to keep my PR to one file to reduce the risk of a conflict with others.  I changed Basic CC too, and there was more than one place that needed >= rather than ==.  I built locally and seem to be OK, but I'm confident that Basic CC code is exercised more than just the issue I was tripping over in SwitchBinary.

All for the better though.